packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1019,9 +1036,16 @@ export class GarbageCollector implements IGarbageCollector {
 
         // If we are running in GC test mode, delete objects for unused routes. This enables testing scenarios
         // involving access to deleted data.
-        if (this.testMode) {
-            this.runtime.updateUnusedRoutes(gcResult.deletedNodeIds);
+        if (this.sweepDataStoresMode) {
```

Comment:
I think we should move these if / else-if blocks to a function called something like handleDeletedRoutes. It can decide one of these things to do based on configs / flags.

Reply:
Sure

Reply:
Because testMode uses `gcResult.deletedNodeIds` instead of `sweepReadyNodes`, I'm opting to not do this in this PR